src/State/HubInterface.php

```diff
@@ -54,8 +54,10 @@ public function popScope(): bool;
      * is automatically removed once the operation finishes or throws.
      *
      * @param callable $callback The callback to be executed
+     *
+     * @return mixed|void The callback's return value, upon successful execution
      */
-    public function withScope(callable $callback): void;
+    public function withScope(callable $callback);
```

Comment:
Please add PHPStan generics to tie the signature of the callable to the return value of this function:

```
@phpstan-template T

@phpstan-param callable(Scope): T $callback

@phpstan-return T
```